### DISCUSSION TITLE
Algolia Search results have no content

### DISCUSSION THREAD

**Discussion by herrtxbias** [facebook/docusaurus#4872] [Q&A]
Hello Everyone,

i've published my docs via netlify and installed there the Netlify Algolia Crawler Plugin. After some configuration i finally managed it to get any results shown in the search modal.

Sadly the only parameters i see in there are the page headings:
![Image](https://cdn.herrtxbias.net/2021-05-30_7ced8618-3ba0-48e7-b290-85df9e677ce1.png)

Does anybody have a clue, what may be wrong with the settings?

**Comment by slorber** [comment]
It is hard to help you if we can't see your site live, it's config or source code, and it's algolia docsearch config (if you even use DocSearch?).

Afaik nobody tested the Netlify system yet and we don't document it yet, so you are in unknown territory here

---

@shortcuts any idea if the Netlify Algolia Crawler Plugin is usable with a Docusaurus site and Docsearch? 

Should we use it ourselves on Docusaurus site and document this workflow? 

What are the advantages? Maybe not wait a scheduled recrawl?

**Reply by shortcuts** [reply]
As long as the records are indexed using the DocSearch helper (provided from the Crawler), it will be usable with DocSearch and Docusaurus!

This is definitely something that could be documented, since we plan to leverage the Crawler with DocSearch in a near future.

- For any DocSearch users, the advantages are mainly the possibility to schedule and trigger a crawl ,but also having a UI that let you create/test your config in real time, out of the box.
- For paid plans, you have a full control on your app, and the possibility to tweak your search results using the other Algolia features.

(In the meantime of the DocSearch migration, if you'd like to see how it is working, feel free to contact me on Discord)

**Reply by slorber** [reply]
Thanks

When you feel it's a good time, maybe we can enable that Netlify plugin on the Docusaurus site itself, and document usage once we get it working?

**Comment by shortcuts** [comment]
The [DocSearch](https://github.com/algolia/docsearch) library follow a specific record structure, which is not the default indexing of the Algolia Crawler.

With the open-source DocSearch crawler, we use [a specific template](https://github.com/algolia/docsearch-configs/blob/master/configs/docusaurus-2.json) for any Docusaurus v2 websites that you can translate with the help of the built-in DocSearch helper in the Algolia Crawler.

The `recordExtractor` in your Crawler config would look like this:
```js
recordExtractor: ({ url, $, contentLength, fileType, helpers }) => {
  return helpers.docsearch({
    selectors: {
      // Algolia Crawler doesn't support xPath, so we rely on cheerio ($) here
      lvl0: $('.menu__link.menu__link--sublist.menu__link--active').text() || $('.navbar__item.navbar__link--active').text(),
      lvl1: ".header h1",
      lvl2: ".article h2",
      lvl3: ".article h3",
      lvl4: ".article h4",
      lvl5: ".article h5",
      content: "article p, article li, article td:last-child",
    },
  });
},
```

Any records in your Algolia index will now have the right format to be used with our front-end library!

Since this is a fairly new feature, If you see any issue with the output from the Crawler, feel free to open an issue on the [DocSearch](https://github.com/algolia/docsearch) repository so we can iterate on it!

**Reply by jan-molak** [reply]
Thanks @shortcuts; your answer put me on the right track. For anyone who stumbles upon this issue in the future, Algolia has now published [Docusarus v2 templates](https://docsearch.algolia.com/docs/templates/#docusaurus-v2-template)
